src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,134 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> temp = tail;
+        Node<T> newNode = new Node<>(temp, value, null);
+        tail = newNode;
+        if (temp == null) {
+            head = newNode;
+        } else {
+            temp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        final Node<T> curNode = getNode(index);
+        final Node<T> prevNode = curNode.prev;
+        final Node<T> newNode = new Node<>(prevNode, value, curNode);
+        if (prevNode == null) {
+            head = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        final Node<T> curNode = getNode(index);
+        final T oldValue = curNode.value;
+        curNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        final Node<T> curNode = getNode(index);
+        final Node<T> next = curNode.next;
+        final Node<T> prev = curNode.prev;
+        if (prev == null) {
+            head = next;
+        } else {
+            prev.next = next;
+        }
+        if (next == null) {
+            tail = prev;
+        } else {
+            next.prev = prev;
+        }
+        size--;
+        return curNode.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        final int index = getIndex(object);
+        if (index == -1) {
+            return false;
+        }
+        remove(index);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return (size == 0);
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", size: " + size);
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        checkIndex(index);
+        Node<T> curNode = head;
+        for (int i = 0; i < index; i++) {
+            curNode = curNode.next;
+        }
```

Comment:
You should decide: from which side it would be faster to reach your index: from head (i++) or from tail (i--)
![image](https://user-images.githubusercontent.com/117547096/212539682-55066cb7-75b5-423b-a4a7-902f8d37d7bf.png)